Source/WebInspectorUI/UserInterface/Models/FontStyles.js

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+WI.FontStyles = class FontStyles
+{
+    constructor(nodeStyles)
+    {
+        this._nodeStyles = nodeStyles;
+
+        this._featuresMap = new Map;
+        this._variationsMap = new Map;
+        this._propertiesMap = new Map;
+
+        this._authoredFontVariationSettingsMap = new Map;
+        this._effectiveWritablePropertyForNameMap = new Map;
+
+        // A significant change happens when there are different variation axes between style changes.
```

Comment:
I guess the comment didn't do its job :) 
But I'm concerned `different variation axis values` adds to the confusion because the next line says that changes to variation axis *values* isn't considered a significant change. Only different axes, identified by their tags, cause a significant change.

Reply:
Reworded like so:
```
// A change in the number of axes or their tags is considered a significant change.
// A change to the value of a known axis is not considered a significant change.
```